dace/frontend/python/replacements.py

```diff
@@ -210,6 +211,165 @@ def eye(pv: 'ProgramVisitor',
+                             external_edges=True)
+
+    return name
+
+
+@oprepo.replaces('numpy.full_like')
+def _numpy_full_like(pv: 'ProgramVisitor',
+                     sdfg: SDFG,
+                     state: SDFGState,
+                     a: str,
+                     fill_value: Number,
+                     dtype: dace.typeclass = None,
+                     shape: Shape = None):
+    """ Creates and array of the same shape and dtype as a and initializes it
+        with the fill value.
+    """
+    if a not in sdfg.arrays.keys():
+        raise mem_parser.DaceSyntaxError(
+            pv, None, "Prototype argument {a} is not SDFG data!".format(a=a))
+    desc = sdfg.arrays[a]
+    dtype = dtype or desc.dtype
+    shape = shape or desc.shape
+    return _numpy_full(pv, sdfg, state, shape, fill_value, dtype)
+
+
+@oprepo.replaces('numpy.ones')
+def _numpy_ones(pv: 'ProgramVisitor',
+                sdfg: SDFG,
+                state: SDFGState,
+                shape: Shape,
+                dtype: dace.typeclass = dace.float64):
+    """ Creates and array of the specified shape and initializes it with ones.
+    """
+    return _numpy_full(pv, sdfg, state, shape, 1.0, dtype)
+
+
+@oprepo.replaces('numpy.ones_like')
+def _numpy_ones_like(pv: 'ProgramVisitor',
+                     sdfg: SDFG,
+                     state: SDFGState,
+                     a: str,
+                     dtype: dace.typeclass = None,
+                     shape: Shape = None):
+    """ Creates and array of the same shape and dtype as a and initializes it
+        with ones.
+    """
+    return _numpy_full_like(pv, sdfg, state, a, 1.0, dtype, shape)
+
+
+@oprepo.replaces('numpy.zeros')
+def _numpy_zeros(pv: 'ProgramVisitor',
+                 sdfg: SDFG,
+                 state: SDFGState,
+                 shape: Shape,
+                 dtype: dace.typeclass = dace.float64):
+    """ Creates and array of the specified shape and initializes it with zeros.
+    """
+    return _numpy_full(pv, sdfg, state, shape, 0.0, dtype)
+
+
+@oprepo.replaces('numpy.zeros_like')
+def _numpy_zeros_like(pv: 'ProgramVisitor',
+                      sdfg: SDFG,
+                      state: SDFGState,
+                      a: str,
+                      dtype: dace.typeclass = None,
+                      shape: Shape = None):
+    """ Creates and array of the same shape and dtype as a and initializes it
+        with zeros.
+    """
+    return _numpy_full_like(pv, sdfg, state, a, 0.0, dtype, shape)
+
+
+@oprepo.replaces('numpy.copy')
+def _numpy_copy(pv: 'ProgramVisitor',
+                sdfg: SDFG,
+                state: SDFGState,
+                a: str):
+    """ Creates a copy of array a.
+    """
+    if a not in sdfg.arrays.keys():
+        raise mem_parser.DaceSyntaxError(
+            pv, None, "Prototype argument {a} is not SDFG data!".format(a=a))
+    # TODO: The whole AddTransientMethod class should be move in replacements.py
```

Comment:
Don't we have `sdfg.add_temp_transient` for that?

Reply:
This methods handles different types of data (scalar, array, stream) The SDFG class method is only for arrays, right?